fsm/fsm.go

```diff
@@ -57,64 +179,88 @@ type FSM struct {
 }
 
 // ExecuteCmd executes a command in the FSM
-func (m *FSM) ExecuteCmd(cmd, txt string, fsmDomain *Domain) (answers []query.Answer, extension string) {
-	var transition TransitionFunc
-	var tuple CmdStateTuple
-
-	previousState := m.State
-
-	tupleFromAny := CmdStateTuple{cmd, -1}
-	tupleNormal := CmdStateTuple{cmd, m.State}
-	tupleCmdAny := CmdStateTuple{"any", m.State}
-
-	if fsmDomain.TransitionTable[tupleFromAny] == nil {
-		if fsmDomain.TransitionTable[tupleCmdAny] == nil {
-			transition = fsmDomain.TransitionTable[tupleNormal] // There is no transition "From Any" with cmd, nor "Cmd Any"
-			tuple = tupleNormal
-		} else {
-			transition = fsmDomain.TransitionTable[tupleCmdAny] // There is a transition "Cmd Any"
-			tuple = tupleCmdAny
+// TODO: Document what the ExecuteCmd does
+func (m *FSM) ExecuteCmd(cmd, question string, fsmDomain *Domain) (answers []query.Answer, extension string) {
```

Comment:
I broke the `ExecuteCmd` down into smaller pieces. It was large and did quite a few things. I think this way it makes it easier to understand.